CMakeLists.txt

```diff
@@ -1018,21 +1018,22 @@ configure_file(
 if(CONFIG_CPU_HAS_MPU AND CONFIG_USERSPACE)
 
     if(CONFIG_MPU_REQUIRES_POWER_OF_TWO_ALIGNMENT AND CONFIG_APP_SHARED_MEM)
-        set(GEN_APP_SMEM $ENV{ZEPHYR_BASE}/scripts/gen_app_smem.py)
+        set(GEN_APP_SMEM $ENV{ZEPHYR_BASE}/scripts/gen_app_partitions.py)
         set(APP_SMEM_LD "${PROJECT_BINARY_DIR}/include/generated/app_smem.ld")
-        set(OBJ_FILE_DIR "${PROJECT_BINARY_DIR}/../")
 
         add_custom_target(
           ${APP_SMEM_DEP} ALL
-          DEPENDS zephyr_prebuilt
+	  DEPENDS app
+	  DEPENDS ${PRIV_STACK_DEP}
```

Comment:
Could you add a CMake comment here that explains how this depends on PRIV_STACK_DEP ?

Reply:
actually that was the only way to make the build to work. When we consider a multi threaded build I was not getting all possible build files because this Cmd will get uncoupled as soon as libapp.a is ready. By adding this I was able to make the build sequential just before prebuilt was done.